### DISCUSSION TITLE
✅ Copy-paste on iOS?

### DISCUSSION THREAD

**Discussion by skye10** [isontheline/pro.webssh.net#609] [Q&A] [answered]
Does WebSSH support copy-paste on iOS?

**Comment by isontheline** [comment]
Yes it does.

Any issue?

**Reply by isontheline** [reply]
However it need to be improved : https://github.com/isontheline/pro.webssh.net/issues/144

**Comment by skye10** [comment]
Yes, I just want to be able to select some text on the shell window, copy
it, and paste it (for example, as an argument to another shell command).
When I click (press) on some text, the menu that pops up is “Export”,
rather than a selector that I can use to select some text and copy it.

Hope what I’m asking for makes sense…
Thank You!

**Reply by isontheline** [reply, marked as answer]
Yes you’re right, yet on iPhone / iPad you need to choose export in order to make a text selection in another popup.

It will be improved by #144 later in order to be the same as macOS

**Comment by jarrah31** [comment]
Could the "Export" popup be renamed to make it more intuitive as to its purpose please? Something like "Export for copy/paste functionality", or "Open Copy/Paste Window"? I too was attempting to copy/paste lines of text today and because it behaves differently to the usual iOS/iPadOS behaviour (hold to select text), I initially thought Export just copied all text into the Copy buffer.

**Reply by isontheline** [reply]
I will rename this menu asap : https://github.com/isontheline/pro.webssh.net/issues/713

**Reply by jarrah31** [reply]
thanks!

**Comment by Plhjk** [comment]


**Reply by isontheline** [reply]
Which ?

**Reply by mbierman** [reply]
I'm afraid I find the copy feature very hard to use for two reasons. 

- I often want to copy a selection of text, not the whole screen. This is better than nothing, but almost always means I have to paste, edit, copy, and paste again. 
- I don't understand what the ui is doing with the floating aspect after long tap. It feels like it is in the way and not what I expect.

**Reply by isontheline** [reply]
You're right, the whole copy/paste feature should be changed : https://github.com/isontheline/pro.webssh.net/issues/722
